src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -39,8 +45,51 @@
      * Андрей - 600
      * София - 900</p>
      */
+    final DateTimeFormatter formatter = DateTimeFormatter.ofPattern("dd.MM.yyyy");
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo)
             throws Exception {
-        return null;
+        LocalDate date1 = LocalDate.parse(dateFrom, formatter);
+        LocalDate date2 = LocalDate.parse(dateTo, formatter);
+        if (date1.isAfter(date2)) {
+            throw new IllegalDateParametersException("Wrong parameters");
+        }
+
+        StringBuilder result = new StringBuilder("Отчёт за период ").append(dateFrom)
+                .append(" - ").append(dateTo);
+        List<EmployeeWorkdaySummary> summaryObjects = buildSummaryObjects(data);
+
+        for (int i = 0; i < names.length; i++) {
+            int salary = 0;
+            for (int j = 0; j < summaryObjects.size(); j++) {
+                if (names[i].equals(summaryObjects.get(j).getPersonName())) {
+                    if (isBetween(date1, date2, summaryObjects.get(j).getDate())) {
```

Comment:
let's move this checking at the previous `if` statement)
because now we have four nested blocks